### ISSUE TITLE
[gvim] Unable to execute

### ISSUE THREAD

**Issue by uberhacker** [skycocker/chromebrew#3072]
With sommelier running:
```
$ gvim
E233: cannot open display
Press ENTER or type command to continue

E852: The child process failed to start the GUI
Press ENTER or type command to continue
```
The same error occurs on all architectures.  @wwwiiilll: Did you want to try and fix this?

Also should note there are no environment variables set such as VIMRUNTIME or LC_ALL.

**Comment by JL2210** [comment]
`sommelier -X gvim`?

**Comment by uberhacker** [comment]
```
$ sommelier -X gvim
Warning: Unknown scale factor: 1111
Warning: Unknown scale factor: 952
Warning: Unknown scale factor: 909
Warning: Unknown scale factor: 870
Warning: Unknown scale factor: 833
Warning: Unknown scale factor: 769
Warning: Unknown scale factor: 667
E233: cannot open display
Press ENTER or type command to continue

E852: The child process failed to start the GUI
Press ENTER or type command to continue
```

**Comment by JL2210** [comment]
`echo GDK_BACKEND=${GDK_BACKEND} && cat ~/.bashrc`, please.

**Comment by uberhacker** [comment]
```
$ echo $GDK_BACKEND
wayland
$ cat ~/.bashrc
---snip---
# Sommelier environment variables + daemon
# See https://github.com/dnschneid/crouton/wiki/Sommelier-(A-more-native-alternative-to-xiwi)
if [ ! -d /tmp/.X11-unix ]; then
mkdir /tmp/.X11-unix
fi
sudo chmod -R 1777 /tmp/.X11-unix
sudo chown root:root /tmp/.X11-unix
alias startsommelier="set -a && source ~/.sommelier.env && set +a && initsommelier"
startsommelier
```

**Comment by JL2210** [comment]
Not helping. I need to see all of it.

**Comment by uberhacker** [comment]
This is the only relevant part that could be affecting the issue.  I already know what the problem is but since you guys didn't listen and were insisting that we build the way that _every_ major distro does it without understanding the impact, this is the result.

**Comment by JL2210** [comment]
I need to see if there are any other commands in there that might overwrite
the environment variables. All of the other GTK+3 apps that I use work.

**Comment by JL2210** [comment]
`echo ${DISPLAY}`

**Comment by uberhacker** [comment]
:0

**Comment by JL2210** [comment]
Have you tried it with `Xwayland :0` running?

**Comment by uberhacker** [comment]
startx and running from xterm produces the same result.

**Comment by JL2210** [comment]
I'm looking into it.

**Comment by JL2210** [comment]
Give this a shot `yes | crew install xhost; xhost +`

**Comment by JL2210** [comment]
OK. Can you please unlock #3059? I may have a solution.

**Comment by uberhacker** [comment]
```
$ xhost +localhost
xhost:  unable to open display ":0"
$ gvim
E233: cannot open display
Press ENTER or type command to continue

E852: The child process failed to start the GUI
Press ENTER or type command to continue
```

**Comment by JL2210** [comment]
... We can probably revert to GTK+ 2, however I can't find a reason why updating would break anything.

**Comment by uberhacker** [comment]
Fixed in #3444.
